make rpm fails due to too many args to open

This happens in when compiling one of the test binaries under `build`, i.e. when running `make rpm`.  
Running just the testsuite with `make check` works fine.  
I first noticed this after I'd upgraded to F33.  
  
    In file included from /usr/include/fcntl.h:329,
                     from dump_core.c:5:
    In function 'open',
        inlined from 'dump_core' at dump_core.c:54:14:
    /usr/include/bits/fcntl2.h:44:5: error: call to '__open_too_many_args' declared with attribute error: open can be called either with 2 or 3 arguments, not more
       44 |     __open_too_many_args ();

--- a/tests/dump_core.c
+++ b/tests/dump_core.c
@@ -2,6 +2,10 @@
 #define _GNU_SOURCE
 #endif
 
+#if defined _FORTIFY_SOURCE
+#undef _FORTIFY_SOURCE
+#endif
+
 #include <fcntl.h>
 #include <stdbool.h>
 #include <stdio.h>

--- a/tests/gdb_frame.c
+++ b/tests/gdb_frame.c
@@ -76,7 +76,7 @@ test_gdb_frame_parse_frame_start(void)
 
     for (int i = 0; i < sizeof (t) / sizeof (*t); i++)
     {
-        int number;
+        uint32_t number;
         char *old_input = t[i].input;
         g_assert_cmpuint(t[i].parsed_char_count, ==, sr_gdb_frame_parse_frame_start((const char **)&t[i].input, &number));
         if (0 < t[i].parsed_char_count)

--- a/tests/js_frame.c
+++ b/tests/js_frame.c
@@ -18,7 +18,7 @@
         } \
         if (    ( e_function && (g_strcmp0(r_frame->function_name, e_function) != 0)) \
              || (!e_function && r_frame->function_name)) { \
-            fprintf(stderr, "function_name -> '%s' != '%s'\n", r_frame->function_name, e_function); \
+            fprintf(stderr, "function_name -> '%s' != '%s'\n", r_frame->function_name, e_function ? (char *)e_function : ""); \
             g_assert_null(i_input); \
         } \
         if (    ( e_file && (g_strcmp0(r_frame->file_name, e_file) != 0)) \

--- a/tests/js_platform.c
+++ b/tests/js_platform.c
@@ -10,7 +10,7 @@
     do { \
         const char *value = sr_js_engine_to_string(a_engine); \
         if (g_strcmp0(value, a_expected) != 0) { \
-            fprintf(stderr, "%s != %s\n", value, a_expected); \
+            fprintf(stderr, "%s != '%s'\n", value, a_expected ? (char *)a_expected : ""); \
             g_assert_true(!"sr_js_engine_to_string(" #a_engine ") != " #a_expected); \
         } \
     } while (0)
@@ -44,7 +44,7 @@ test_js_engine_from_string(void)
     do { \
         const char *t_value = sr_js_runtime_to_string(a_runtime); \
         if (g_strcmp0(t_value, a_expected) != 0) { \
-            fprintf(stderr, "%s != %s\n", t_value, a_expected); \
+            fprintf(stderr, "%s != '%s'\n", t_value, a_expected ? (char *)a_expected : ""); \
             g_assert_true(!"sr_js_runtime_to_string(" #a_runtime ") != " #a_expected); \
         } \
     } while (0)
@@ -78,7 +78,7 @@ static void
 test_js_platform_basics(void)
 {
     sr_js_platform_t platform;
-    g_assert_true(platform = sr_js_platform_new());
+    g_assert_true((platform = sr_js_platform_new()));
 
     sr_js_platform_init(platform, SR_JS_RUNTIME_NODEJS, SR_JS_ENGINE_V8);
 
@@ -86,7 +86,7 @@ test_js_platform_basics(void)
     g_assert_true(sr_js_platform_runtime(platform) == SR_JS_RUNTIME_NODEJS);
 
     sr_js_platform_t another_platform;
-    g_assert_true(another_platform = sr_js_platform_dup(platform));
+    g_assert_true((another_platform = sr_js_platform_dup(platform)));
 
     /* In the future we might need also the above test but now the test would
      * fail because sr_js_platform_t is a pure integer.
@@ -130,7 +130,7 @@ test_js_platform_basics(void)
         } \
         else { \
             if (g_strcmp0(r_error_message, e_error_message) != 0) { \
-                fprintf(stderr, "%s != %s\n", r_error_message, e_error_message); \
+                fprintf(stderr, "%s != %s\n", r_error_message, e_error_message ? (char *)e_error_message : ""); \
                 g_assert_true(!"Got invalid error message: sr_js_platform_from_string(" \
                         #i_runtime_str "," #i_version_str ")"); \
             } \

--- a/tests/js_stacktrace.c
+++ b/tests/js_stacktrace.c
@@ -312,7 +312,7 @@ test_js_stacktrace_to_json(void)
         }
         
         const size_t elen = strlen(expected);
-        if (expected[elen - 1] = '\n') {
+        if ((expected[elen - 1] = '\n')) {
             expected[elen - 1] = '\0';
         }
         
@@ -375,7 +375,7 @@ test_js_stacktrace_from_json(void)
         }
         
         if (0 != g_strcmp0(stacktrace1->exception_name, stacktrace2->exception_name)) {
-            fprintf(stderr, "'%s': exception_name -> '%s' != '%s'\n", stacktrace1->exception_name, stacktrace2->exception_name);
+            fprintf(stderr, "test_js_stacktrace_from_json: exception_name -> '%s' != '%s'\n", stacktrace1->exception_name, stacktrace2->exception_name);
             g_assert_true(!"Error names do no match");
         }
         

--- a/tests/normalize.c
+++ b/tests/normalize.c
@@ -51,7 +51,6 @@ static void
 test_normalize_gdb_thread(void)
 {
     struct sr_gdb_thread *threads[3];
-    struct sr_gdb_frame *frame;
 
     threads[0] = create_thread(5, "aa", "abort", "raise", "__assert_fail_base", "ee");
 

--- a/tests/report.c
+++ b/tests/report.c
@@ -88,7 +88,7 @@ test_report_add_auth(void)
     report_iterator = report->auth_entries;
 
     for (size_t i = 0;
-         i < G_N_ELEMENTS(expected), NULL != report_iterator;
+         i < G_N_ELEMENTS(expected) && NULL != report_iterator;
          i++, report_iterator = report_iterator->next)
     {
         g_assert_cmpstr(report_iterator->key, ==, expected[i][0]);
@@ -115,7 +115,7 @@ test_report_add_auth(void)
     report_iterator = report->auth_entries;
 
     for (size_t i = 0;
-         i < G_N_ELEMENTS(expected), NULL != report_iterator;
+         i < G_N_ELEMENTS(expected) && NULL != report_iterator;
          i++, report_iterator = report_iterator->next)
     {
         g_assert_cmpstr(report_iterator->key, ==, expected[i][0]);

--- a/tests/utils.c
+++ b/tests/utils.c
@@ -258,7 +258,7 @@ test_parse_hexadecimal_0xuint64(void)
     g_assert_cmpint(11, ==, sr_parse_hexadecimal_0xuint64((const char **)&input, &num));
     g_assert_cmpint(*input, ==, ' ');
     g_assert_cmpuint(num, ==, 0xfafaffff);
-    *input++;
+    input++;
     g_assert_cmpint(18, ==, sr_parse_hexadecimal_0xuint64((const char **)&input, &num));
     g_assert_cmpint(*input, ==, '\0');
     g_assert_cmpuint(num, ==, 0x2badf00dbaadf00d);
@@ -277,7 +277,7 @@ test_indent(void)
 
 void do_check(char **in, size_t inlen, char **out, size_t outlen)
 {
-    printf("%d %d\n", inlen, outlen);
+    printf("%ld %ld\n", inlen, outlen);
     sr_struniq(in, &inlen);
     g_assert_true(inlen == outlen);
     

